pkg/daemons/agent/agent.go

```diff
@@ -29,12 +29,15 @@ func Agent(config *config.Agent) error {
 	defer logs.FlushLogs()
 
 	startKubelet(config)
-	startKubeProxy(config)
+
+	if !config.DisableKubeProxy {
+		return startKubeProxy(config)
+	}
 
 	return nil
 }
 
-func startKubeProxy(cfg *config.Agent) {
+func startKubeProxy(cfg *config.Agent) error {
```

Comment:
Is the return signature necessary since we never actually return if there is a problem (I'm referring to the call to `logrus.Fatalf`)

Reply:
It's a side effect of a follow on PR.  I tried to break this up into smaller commit.  I'll remove this as it's not needed for this. Thanks!